addon/components/mobiledoc-editor/component.js

```diff
@@ -302,23 +341,39 @@ export default Component.extend({
     const markupTags = arrayToMap(editor.activeMarkups.map(m => m.tagName));
     // editor.activeSections are leaf sections.
     // Map parent section tag names (e.g. 'p', 'ul', 'ol') so that list buttons
-    // are updated.
+    // can be bound.
+    // Also build a map of section attributes for the same reason.
     let sectionParentTagNames = editor.activeSections.map(s => {
       return s.isNested ? s.parent.tagName : s.tagName;
     });
     const sectionTags = arrayToMap(sectionParentTagNames);
+    const sectionAttributes = {};
+    editor.activeSections.forEach(s => {
```

Comment:
This whole `inputModeDidChange` method is pretty complex. Maybe at least convert the anonymous function into something named?